hs/src/Reach/AST/DLBase.hs

```diff
@@ -13,6 +13,14 @@ import Reach.Counter
 import Reach.Pretty
 import Reach.Texty
 import Reach.Util
+import Control.Monad.Identity
+import Control.Monad.Reader
+import Data.Functor ((<&>))
+
+type PrettyT = ReaderT (M.Map DLVar Doc) Identity
```

Comment:
This would be named `PrettyApp` if we're following our naming convention, although probably `PrettySubstApp` is more accurate. We'd also give the `M.Map` a new name `Env`